weather_dl/download_pipeline/parsers.py

```diff
@@ -441,13 +471,19 @@ def require(condition: bool, message: str, error_type: t.Type[Exception] = Value
         if not isinstance(selection[key], list):
             selection[key] = [selection[key]]
 
-    return Config.from_dict(config)
+    config = Config.from_dict(config)
+
+    _validate_target_path(config)
+
+    return config
 
 
-def prepare_target_name(config: Config) -> str:
+def prepare_target_name(partition_config: Config) -> str:
     """Returns name of target location."""
-    partition_dict = OrderedDict((key, typecast(key, config.selection[key][0])) for key in config.partition_keys)
-    target = config.target_path.format(*partition_dict.values(), **partition_dict)
+    partition_dict = OrderedDict(
+        (key, typecast(key, partition_config.selection[key][0]))
+        for key in partition_config.partition_keys)
```

Comment:
I'm ok with making this multiline, but then let's go all the way :)
```suggestion
    partition_dict = OrderedDict(
        (key, typecast(key, partition_config.selection[key][0]))
        for key in partition_config.partition_keys
    )
```